### PR TITLE
Add availability to nonScalar for recursive Array and noSQL usage

### PR DESCRIPTION
**Pull request by bsormagec**
If you are using NoSQL you need this. Because noSQL has ability to recursive array or objects and you can save them but in this package its returning null always changes. With this change you can enable it from config. Thanks.

`{
"id":"1",
"name": "test",
"surname": "user",
"settings": {
"hasCar": true,
"hasBike": false,
"isParent": false,
},
"isActive":true
}`

In this json example when i change settings object now this package not check changes. but with my changes if anybody wanna use this only enable it from settings.

### CODE REVIEW

File: config/audit.php [hunk 1]
```diff
@@ -23,7 +23,7 @@
     |
     */
 
-    'implementation' => OwenIt\Auditing\Models\Audit::class,
+    'implementation' => App\Models\Audit::class,
```
Comment (quetzyg) [inline config/audit.php]: Why was this changed?

File: config/audit.php [hunk 2]
```diff
@@ -36,7 +36,7 @@
 
     'user' => [
         'morph_prefix' => 'user',
-        'guards'       => [
+        'guards' => [
```
Comment (quetzyg) [inline config/audit.php]: Lets leave the array alignment alone, please.

File: config/audit.php [hunk 3]
```diff
@@ -105,7 +105,7 @@
     |
     */
 
-    'threshold' => 0,
+    'threshold' => 100,
```
Comment (quetzyg) [inline config/audit.php]: Put it back as it was.

File: config/audit.php [hunk 4]
```diff
@@ -140,8 +140,8 @@
 
     'drivers' => [
         'database' => [
-            'table'      => 'audits',
```
Comment (quetzyg) [inline config/audit.php]: Change it back.

File: config/audit.php [hunk 5]
```diff
@@ -155,4 +155,16 @@
...
+    /*
+    |--------------------------------------------------------------------------
+    | Audit Allow Saving NonScalar Values and Recursive Arrays
+    |--------------------------------------------------------------------------
+    |
+    | Allow Saving NonScalar Values and Recursive Arrays
+    |
+    */
+
+    'nonScalar' => true,
```
Comment (quetzyg) [inline config/audit.php]: Follow the configuration index convention and use `snake_case`.

File: config/audit.php [hunk 6]
```diff
@@ -141,7 +141,7 @@
     'drivers' => [
         'database' => [
             'table'      => 'audits',
-            'connection' => null,
+            'connection' => 'mongodb',
```
Comment (quetzyg) [inline config/audit.php]: Put it back.

File: config/audit.php [hunk 7]
```diff
@@ -155,4 +155,15 @@
     */
 
     'console' => false,
+
+    /*
+    |--------------------------------------------------------------------------
```
Comment (quetzyg) [inline config/audit.php]: What's with the double header?

File: config/audit.php (hunk 6 above)
Comment (bsormagec) [reply config/audit.php]: Im fixing all back 🐙

File: src/Auditable.php [hunk 8]
```diff
@@ -103,13 +103,15 @@ protected function resolveAuditExclusions()
...
-            // Apart from null, non scalar values will be excluded
-            if (is_array($value) || (is_object($value) && !method_exists($value, '__toString'))) {
-                $this->excludedAttributes[] = $attribute;
+        if (!Config::get('audit.nonScalar', false)) {
+            // Valid attributes are all those that made it out of the exclusion array
+            $attributes = array_except($this->attributes, $this->excludedAttributes);
+
+            foreach ($attributes as $attribute => $value) {
+                // Apart from null, non scalar values will be excluded
+                if (\is_array($value) || (\is_object($value) && !method_exists($value, '__toString'))) {
```
Comment (quetzyg) [inline src/Auditable.php]: Also, remove the backslash from `is_array()` and `is_object()`.

Comment (bsormagec) [reply src/Auditable.php]: Sure...

File: config/audit.php (hunk 5 above)
Comment (bsormagec) [reply config/audit.php]: fixed

File: config/audit.php (hunk 4 above)
Comment (bsormagec) [reply config/audit.php]: fixed

File: config/audit.php (hunk 3 above)
Comment (bsormagec) [reply config/audit.php]: fixed

File: config/audit.php (hunk 2 above)
Comment (bsormagec) [reply config/audit.php]: fixed

**Comment by quetzyg** [comment]
>If you are using NoSQL you need this. Because noSQL has ability to recursive array or objects and you can save them but in this package its returning null always changes. With this change you can enable it from config. Thanks.

Please provide an example of what the actual issue is, and how this change fixes it.

**Comment by bsormagec** [comment]
`{
"id":"1",
"name": "test",
"surname": "user",
"settings": {
"hasCar": true,
"hasBike": false,
"isParent": false,
},
"isActive":true
}`

In this json example when i change settings object now this package not check changes. but with my changes if anybody wanna use this only enable it from settings.

**Comment by quetzyg** [comment]
In order to go forward with this, you'll have to add tests for the use case you're describing.

Test to pass, test to fail, every scenario.

**Comment by quetzyg** [comment]
Closing due to lack of updates
